Selenium_Basics/Selenium_Basics/EpamSearchTest.cs

```diff
@@ -0,0 +1,111 @@
+﻿using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Selenium_Basics
+{
+    public class EpamSearchTest
+    {
+        private IWebDriver _driver { get; set; }
+        private const string EpamHomePageUrl = "https://www.epam.com/";
+
+
+
+        [SetUp]
+        public void SetupBrowser()
+        {
+            _driver = new ChromeDriver();
+            _driver.Manage().Window.Maximize();
+            _driver.Manage().Timeouts().PageLoad = TimeSpan.FromSeconds(20);
+            _driver.Navigate().GoToUrl(EpamHomePageUrl);
+            AcceptCookiePolicyIfPresent();
+        }
+
+        [Test]
+        public void VerifyThatSearchResultsCorrespondToSearchCriteria()
+        {
+
+            // Step 1. Click on the Search button in the Header
+            var searchButton = _driver.FindElement(By.XPath("//span[contains(@class,'dark-iconheader-search__search-icon')]"));
+            searchButton.Click();
+
+            // Step 2. Type word "Automation" in the Search input field
+            string WordToSearch = "Automation";
+            var searchInput = _driver.FindElement(By.Id("new_form_search"));
+            searchInput.SendKeys(WordToSearch);
+
+            // Step 3. Click on the 'Find' button
+            var findButton = _driver.FindElement(By.XPath("//*[@class='bth-text-layer']"));
+            findButton.Click();
+            
+
+            // Step 4. Check that the search page is opened with the correct URL
+            Assert.AreEqual($"https://www.epam.com/search?q={WordToSearch}", _driver.Url);
```

Comment:
please review clarification 4 in HW file. There should be no constants inside asserts!
please move text constants before all actions in test as 
var a = "x";
var b = "y":
var element = _driver.....

Reply:
Also please ass message in assert